helix-core/src/main/java/org/apache/helix/controller/stages/BestPossibleStateCalcStage.java

```diff
@@ -90,60 +96,86 @@ private BestPossibleStateOutput compute(ClusterEvent event, Map<String, Resource
 
     BestPossibleStateOutput output = new BestPossibleStateOutput();
 
-    for (String resourceName : resourceMap.keySet()) {
-      logger.debug("Processing resource:" + resourceName);
+    // Reset current INIT/RUNNING tasks on participants for throttling
+    JobRebalancer.resetActiveTaskCount(cache.getLiveInstances().keySet(), currentStateOutput);
 
-      Resource resource = resourceMap.get(resourceName);
-      // Ideal state may be gone. In that case we need to get the state model name
-      // from the current state
-      IdealState idealState = cache.getIdealState(resourceName);
+    PriorityQueue<JobResourcePriority> jobResourceQueue = new PriorityQueue<JobResourcePriority>();
```

Comment:
SortedSet is an interface, so you mean using TreeSet?
Why do you think that will be better? I think there is no big difference between a TreeSet and PriorityQueue here.

Besides, we do want to implement some kind of job priority here, as you can see. So PriorityQueue is easier for others to understand.